lib/Transforms/ConvertIfToSelect/ConvertIfToSelect.cpp

```diff
@@ -88,20 +89,16 @@ struct IfToSelectConversion : OpRewritePattern<scf::IfOp> {
             ifOp.getLoc(), cond, trueVal, falseVal);
       }
 
-      // Update the secretness of the new results, using the "secretness" of
-      // the condition which could have been either "secret" or "uninitialized"
-      for (auto &r : newResults) {
-        auto *lattice = solver->getOrCreateState<SecretnessLattice>(r);
-        solver->propagateIfChanged(lattice, lattice->join(secretness));
-      }
-
       rewriter.replaceOp(ifOp, newResults);
     }
 
-    return success();
+    // re-run the analysis propagate secretness across
+    // newly created Values
+    return solver->initializeAndRun(top);
   }
 
  private:
+  Operation *top;
```

Comment:
Is there a potential for this to become null during a pass, e.g., because a rewrite replaces the main function operation the pass acts on?

Reply:
this is often the ModuleOp, which the pass takes as the root operation (runOnOperation) and is not allowed to alter it.

https://mlir.llvm.org/docs/PassManagement/

> All passes in MLIR derive from OperationPass often (ModuleOp) and adhere to the following restrictions
>
> Must not modify the state of operations other than the operations that are nested under the current operation.

Reply:
Oh, you're right - since you set this to `getOperation()` in the `runOnOperation` function, it would indeed be a bug for a pass to delete that op! Can you add a comment about that? Thanks!